keydata_test.go

```diff
@@ -1369,12 +1304,14 @@ func (s *keyDataSuite) TestReadAndWriteWithUnsaltedKeyDigest(c *C) {
 	c.Check(ok, testutil.IsTrue)
 }
 
+// Legacy tests
```

Comment:
Should the comment here be above `TestReadAndWriteWithUnsaltedKeyDigest`?

Reply:
Yes, moved it. Also I see that the `TestReadKeyData*` tests make use of `SetAuthorizedSnapModels` which will now be legacy behavior so I will move it above those tests too in the next PR.